Update-requirements

kjaymiller opened the pull request:
updates requirements to have pip-tools and gh_issues... Also updates those requirements to the latest version

kjaymiller committed:
update requirements

kjaymiller committed:
Merge branch 'main' of github.com:Python-Community-News/Site

jonafato commented on requirements.in:
```diff
@@ -1,3 +1,5 @@
 render_engine
 typer
+pip-tools
+gh-issues
```
This is already present, but I believe it needs to be updated with `pip-compile -P gh-issues` or `pip-compile -U`.

Edit: we actually have a pre-commit task defined for this. You can run it manually with `pre-commit run --hook-stage manual --all-files`.

jonafato commented on requirements.in:
```diff
@@ -1,3 +1,5 @@
 gh_issues
 render_engine
 typer
+pip-tools
```
I'm -1 on including `pip-tools` in the requirements file here. In general, I think we should be relying on the task defined in our pre-commit config for this.

kjaymiller committed:
removes pip-tools and renames dev --> automate

kjaymiller committed:
run pre-commit

jonafato committed:
Be more explicit about split requirements files

These requirements files may need to be organized in directories at some
point, but for now, this change ensures that they're manged individually
by pre-commit and handle inheritence consistently.
